datafusion/physical-expr/src/expressions/dynamic_filters.rs

```diff
@@ -36,16 +36,8 @@ use super::Column;
 /// A dynamic [`PhysicalExpr`] that can be updated by anyone with a reference to it.
 #[derive(Debug)]
 pub struct DynamicFilterPhysicalExpr {
-    /// The original children of this PhysicalExpr, if any.
-    /// This is necessary because the dynamic filter may be initialized with a placeholder (e.g. `lit(true)`)
-    /// and later remapped to the actual expressions that are being filtered.
-    /// But we need to know the children (e.g. columns referenced in the expression) ahead of time to evaluate the expression correctly.
-    // columns: Vec<Arc<dyn PhysicalExpr>>,
-    // /// If any of the children were remapped / modified (e.g. to adjust for projections) we need to keep track of the new children
-    // /// so that when we update `current()` in subsequent iterations we can re-apply the replacements.
-    remapped_schema: Option<SchemaRef>,
     /// The source of dynamic filters.
-    inner: Arc<RwLock<PhysicalExprRef>>,
+    inner: PhysicalExprRef,
```

Comment:
@jayzhan211 how can this have multiple readers and a writer updating with some sort of write lock?

Reply:
I think we don't need it. Given a source filter, you create snapshot with the schema. Then you evaluate based on the remapped filter. When you need a new source filter, instead of updating it, just create a new one

Reply:
But how do you pipe the new filter down into other operators?

The whole point is that you can create a filter at planning time, bind it to a ParquetSource and a SortExec (for example) and then the SortExec can dynamically update it at runtime.